### PR TITLE
REL: 20.1.0

### LINKED ISSUE #2142: Inconsistent header/data information in aparcaseg
version: 20.1.0 master (as of patch for #2138)

I'm getting odd behavior loading in any of the functional space `aparcaseg` files:

```python
from nilearn.image import load_img, new_img_like
img = load_img('GLS005_ses-2_task-graphlocalizer_space-MNI152NLin2009cAsym_desc-aparcaseg_dseg.nii.gz')
print('From fmriprep:')
print(f'data type: {img.get_data_dtype()}\n')

print('Unique values:')
with np.printoptions(precision=3, suppress=True):
    print(np.unique(img.dataobj))
    
img2 = new_img_like(x, np.round(data).astype('<i2'))
img2.to_filename('img2.nii.gz')
img2 = load_img('img2.nii.gz')
print('After Resaving:')
print(f'data type: {img2.get_data_dtype()}\n')
print('Unique values:')
with np.printoptions(precision=3, suppress=True):
    print(np.unique(img2.dataobj))
```

```
From fmriprep:
data type: int16

Unique values:
[   0.       1.987    4.006    4.999    6.987    8.011    9.999   10.992
   11.986   13.011   14.005   14.998   15.992   16.986   18.01    24.003
   25.991   28.009   30.99    40.989   43.007   44.001   45.988   47.013
   49.      49.994   50.988   52.012   53.006   54.      58.005   59.993
   62.011   63.005   77.009   84.99   250.994  251.988  253.013  254.006
  255.    1000.002 1000.996 1001.989 1003.014 1005.002 1005.995 1006.989
 1008.014 1009.007 1010.001 1010.995 1011.988 1013.013 1014.007 1015.
 1015.994 1016.988 1018.012 1019.006 1020.    1020.993 1021.987 1023.012
 1024.005 1024.999 1025.993 1026.986 1028.011 1029.005 1029.998 1030.992
 1031.986 1033.011 1034.004 1034.998 2000.004 2000.998 2001.992 2002.985
 2005.004 2005.997 2006.991 2007.985 2009.009 2010.003 2010.997 2011.99
 2013.015 2014.009 2015.002 2015.996 2016.99  2018.014 2019.008 2020.002
 2020.995 2021.989 2023.014 2024.008 2025.001 2025.995 2026.989 2028.013
 2029.007 2030.001 2030.994 2031.988 2033.013 2034.006 2035.   ]

After Resaving:
data type: int16

Unique values:
[   0    2    4    5    7    8   10   11   12   13   14   15   16   17
   18   24   26   28   31   41   43   44   46   47   49   50   51   52
   53   54   58   60   62   63   77   85  251  252  253  254  255 1000
 1001 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015
 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029
 1030 1031 1032 1033 1034 1035 2000 2001 2002 2003 2005 2006 2007 2008
 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022
 2023 2024 2025 2026 2027 2028 2029 2030 2031 2032 2033 2034 2035]
```

I get identical behavior viewing the file in `fsleyes`. I'm not sure exactly what's going on here, but my suspicion is an issue with header/data consistency in some of the generated files. It looks like these were changed to int16 from earlier versions?

In 20.0.7:
```
> fslhd old_space-MNI152NLin2009cAsym_desc-aparcaseg_dseg.nii.gz
sizeof_hdr	348
data_type	FLOAT32
dim0		3
dim1		97
dim2		115
dim3		97
dim4		1
dim5		1
dim6		1
dim7		1
vox_units	mm
time_units	Unknown
datatype	16
nbyper		4
bitpix		32
pixdim0		1.000000
pixdim1		2.000000
pixdim2		2.000000
pixdim3		2.000000
pixdim4		0.000000
pixdim5		0.000000
pixdim6		0.000000
pixdim7		0.000000
vox_offset	352
cal_max		0.000000
cal_min		0.000000
scl_slope	1.000000
scl_inter	0.000000
phase_dim	0
freq_dim	0
slice_dim	0
slice_name	Unknown
slice_code	0
slice_start	0
slice_end	0
slice_duration	0.000000
toffset		0.000000
intent		Unknown
intent_code	0
intent_name
intent_p1	0.000000
intent_p2	0.000000
intent_p3	0.000000
qform_name	MNI_152
qform_code	4
qto_xyz:1	2.000000 0.000000 0.000000 -96.500000
qto_xyz:2	0.000000 2.000000 0.000000 -132.500000
qto_xyz:3	0.000000 0.000000 2.000000 -78.500000
qto_xyz:4	0.000000 0.000000 0.000000 1.000000
qform_xorient	Left-to-Right
qform_yorient	Posterior-to-Anterior
qform_zorient	Inferior-to-Superior
sform_name	MNI_152
sform_code	4
sto_xyz:1	2.000000 0.000000 0.000000 -96.500000
sto_xyz:2	0.000000 2.000000 0.000000 -132.500000
sto_xyz:3	0.000000 0.000000 2.000000 -78.500000
sto_xyz:4	0.000000 0.000000 0.000000 1.000000
sform_xorient	Left-to-Right
sform_yorient	Posterior-to-Anterior
sform_zorient	Inferior-to-Superior
file_type	NIFTI-1+
file_code	1
descrip		xform matrices modified by FixHeaderApplyTransforms (niworkflows v1.1.12).
aux_file
```

In 20.1.0:
```
> fslhd new_desc-aparcaseg_dseg.nii.gz

sizeof_hdr	348
data_type	INT16
dim0		3
dim1		81
dim2		96
dim3		81
dim4		1
dim5		1
dim6		1
dim7		1
vox_units	mm
time_units	Unknown
datatype	4
nbyper		2
bitpix		16
pixdim0		1.000000
pixdim1		2.400000
pixdim2		2.400000
pixdim3		2.400000
pixdim4		0.000000
pixdim5		0.000000
pixdim6		0.000000
pixdim7		0.000000
vox_offset	352
cal_max		0.000000
cal_min		0.000000
scl_slope	0.031052
scl_inter	1017.515503
phase_dim	0
freq_dim	0
slice_dim	0
slice_name	Unknown
slice_code	0
slice_start	0
slice_end	0
slice_duration	0.000000
toffset		0.000000
intent		Unknown
intent_code	0
intent_name
intent_p1	0.000000
intent_p2	0.000000
intent_p3	0.000000
qform_name	MNI_152
qform_code	4
qto_xyz:1	2.400000 0.000000 0.000000 -96.500000
qto_xyz:2	0.000000 2.400000 0.000000 -132.500000
qto_xyz:3	0.000000 0.000000 2.400000 -78.500000
qto_xyz:4	0.000000 0.000000 0.000000 1.000000
qform_xorient	Left-to-Right
qform_yorient	Posterior-to-Anterior
qform_zorient	Inferior-to-Superior
sform_name	MNI_152
sform_code	4
sto_xyz:1	2.400000 0.000000 0.000000 -96.500000
sto_xyz:2	0.000000 2.400000 0.000000 -132.500000
sto_xyz:3	0.000000 0.000000 2.400000 -78.500000
sto_xyz:4	0.000000 0.000000 0.000000 1.000000
sform_xorient	Left-to-Right
sform_yorient	Posterior-to-Anterior
sform_zorient	Inferior-to-Superior
file_type	NIFTI-1+
file_code	1
descrip		xform matrices modified by FixHeaderApplyTransforms (niworkflows v1.2.0+2.g9b0f6
aux_file
```

It's just an extra step to round them back to integers, but seems like something's off here.

Edit:
Just wanted to note that while the output from previous versions was float32, the values were set to the correct integer values, unlike here.

### PR DESCRIPTION
**Pull request by mgxd**
This PR is preparation for the 20.1.0 release. Originally meant to be a long-term support (LTS) release, the LTS tag has shifted to a future release, 20.2.0

Closes #2142

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -18,6 +35,9 @@ distinguish which options were selected for each participant, along with other e
 .. _`nipreps/niworkflows#494`: https://github.com/nipreps/niworkflows/pull/494
 .. _`nipreps/niworkflows#492`: https://github.com/nipreps/niworkflows/pull/492
 
+As with all minor version increments, working directories from previous versions should not be reused. Thank you for using *fMRIPrep*, if you
+encounter any issues with this release, please let us know by posting an issue on our github page!
+
```
Comment (oesteban) [inline CHANGES.rst:40]: Move above
```suggestion
```

File: CHANGES.rst [hunk 2]
```diff
@@ -1,15 +1,32 @@
...
+The second minor release of 2020 is finally here! *fMRIPrep* 20.1.0 introduces a magnitude of new features
+and improvements. Some key additions in this release include:
 
+- A dedicated configuration module to centralize *fMRIPrep*'s many options. Every invocation will now produce a config file
+with the specific options used, located at ``<output-directory>/fmriprep/<subject>/<run-uuid>/fmriprep.toml``.
+- A command-line option (``--anat-derivatives``) to allow skipping anatomical processing by reusing existing anatomical derivatives.
+This can significantly speed up longitudinal study preprocessing.
+- A change in output CIFTI2 subcortical volume orientation to be compatible with HCP Pipeline tools and data.
+
+A full list of changes can be found below.
```
Comment (oesteban) [inline CHANGES.rst:12]:
```suggestion
As with all minor version increments, working directories from previous versions should not be reused.

Thank you for using *fMRIPrep*! If you encounter any issues with this release, please let us know by posting an issue on our GitHub page!

A full list of changes can be found below.
With thanks to Basile Pinsard for contributions.
```

File: CHANGES.rst [hunk 3]
```diff
@@ -1,18 +1,61 @@
...
+  <https://openneuro.org/datasets/ds000031>`__.
+- A change in output CIFTI2 subcortical volume orientation to be compatible with HCP Pipeline tools and data.
+
+As with all minor version increments, working directories from previous versions should not be reused.
+
+Thank you for using *fMRIPrep*! If you encounter any issues with this release, please let us know by posting an issue on our GitHub page!
+
+A full list of changes can be found below.
+With thanks to Basile Pinsard for contributions.
+
```
Comment (oesteban) [inline CHANGES.rst:39]:
```suggestion

* FIX: Dismiss ``echo`` entity on several derivatives and figures outputs (#2133)
```

File: CHANGES.rst [hunk 4]
```diff
@@ -1,18 +1,62 @@
...
+Originally nominated to become the first LTS (long-term support) version of *fMRIPrep*, this release has been supercharged with many new features and bug-fixes.
+To ensure long-term stability, we have postponed the LTS nomination to 20.2 to allow us unlocking the 20.1 earlier, and a more extensive stress testing of the series before jumping into a longer support commitment.
+Some key additions in this release include:
+
+- A centralized `configuration module
+  <https://fmriprep.readthedocs.io/en/latest/api.html#module-fmriprep.config>`__
+  keeping track of *fMRIPrep*'s many options and run-time and environmental
+  circumstances and settings.
+  The new config module, which has been also propagated to other *NiPreps*
+  (`dMRIPrep <https://nipreps.com/dmriprep>`__,
```
Comment (oesteban) [inline CHANGES.rst:15]:
```suggestion
  (`dMRIPrep <https://nipreps.org/dmriprep>`__,
```

**Comment by oesteban** [comment]
@marcbue and @jbwexler have contributed to this release, but their names are not in any of the `.maint/` files.

Please add yourselves (or request to be added), on the alphabetical order of your last name, to either the [`.maint/contributors.json` file](https://github.com/poldracklab/fmriprep/blob/master/.maint/contributors.json) if you want to keep involved with activities around *fMRIPrep* or the [`.maint/former.json` file](https://github.com/poldracklab/fmriprep/blob/master/.maint/former.json) if you prefer us not to bug you from time to time. Status in one or another list can change whenever you'd like.

**Comment by jbwexler** [comment]
Thanks, I'm happy to be added to contributors.json. Is there a way for you or me to edit it directly? If not, I can send a PR.


    "affiliation": "Department of Psychology, Stanford University",
    "name": "Wexler, Joseph",
    "orcid": "0000-0001-6424-8740"

**Comment by oesteban** [comment]
Added. I'll rerun the zenodo revision later today

**Comment by jbwexler** [comment]
Thanks!

**Comment by oesteban** [comment]
@effigies @poldrack @mgxd your feedback on these release notes (i.e., those corresponding to 20.1.0 in https://14066-53175327-gh.circle-artifacts.com/0/tmp/src/fmriprep/docs/_build/html/changes.html#may-25-2020) would be very much appreciated if time permits!

(feedback can be given as suggestions to this PR or slack, email, whatever is convenient, Russ)

I'll be testing the RC-3 I just cut, today. Then early tomorrow release 20.1.0 if nothing important emerged.

**Comment by poldrack** [comment]
looks reasonable to me

File: .mailmap [hunk 5]
```diff
@@ -23,6 +23,9 @@ James D. Kent <james-kent@uiowa.edu>
 James D. Kent <james-kent@uiowa.edu> <jamesdkent21@gmail.com>
 James D. Kent <james-kent@uiowa.edu> Fred Mertz <mertzf@bargle.argle>
 Jarod Roland <jarod.roland@gmail.com>
+Joshep B. Wexler <jbwexler@hushmail.com>
```
Comment (mgxd) [inline .mailmap:26]:
```suggestion
Joseph B. Wexler <jbwexler@hushmail.com>
```

File: .mailmap [hunk 6]
```diff
@@ -23,6 +23,9 @@ James D. Kent <james-kent@uiowa.edu>
...
+Joshep B. Wexler <jbwexler@hushmail.com> Joe B. Wexler
```
Comment (mgxd) [inline .mailmap:27]:
```suggestion
Joseph B. Wexler <jbwexler@hushmail.com> Joe B. Wexler
```

File: .mailmap [hunk 7]
```diff
@@ -23,6 +23,9 @@ James D. Kent <james-kent@uiowa.edu>
...
+Joshep B. Wexler <jbwexler@hushmail.com> jbwexler
```
Comment (mgxd) [inline .mailmap:28]:
```suggestion
Joseph B. Wexler <jbwexler@hushmail.com> jbwexler
```

**Comment by effigies** [comment]
I need to verify that syn-sdc is working...

**Comment by oesteban** [comment]
> I need to verify that syn-sdc is working...

Running on ds031 right now, which has some fieldmaps and also runs without fieldmaps (and no intendedfor field). That should also tell us about the ``--anat-derivatives``.

I'll also run later today on frontera the HBN dataset (the test dataset they are using at Childmind Institute for the reliability). That doesn't have fieldmaps - will check ``--use-syn-sdc`` there too.

**Comment by oesteban** [comment]
@effigies the HBN-1 dataset ran smoothly with ``--use-syn-sdc``. The result is not great (I'd rather not use the flag for this particular dataset), but it did run. It is also true that there are no fieldmaps. We'll be able to check whether the heuristics are working okay with ds031, but that one will take a little while to process.

File: CHANGES.rst [hunk 8]
```diff
@@ -1,25 +1,134 @@
...
+  keeping track of *fMRIPrep*'s many options and run-time and environmental
+  circumstances and settings.
+  The new config module, which has been also propagated to other *NiPreps*
+  (`dMRIPrep <https://nipreps.org/dmriprep>`__,
+  `MRIQC <https://mriqc.readthedocs.io/>`__),
+  comes to robustify the run-to-run replicability of *fMRIPrep* (e.g., tracking random seeds),
+  make the option handling more modular but consistent (e.g., setting the ground for a
+  command-line interface built off of the config module),
+  and ease troubleshooting and telemetry.
+- The *anatomical preprocessing fast-track*: a new command-line option
```
Comment (mgxd) [inline CHANGES.rst:21]:
```suggestion
- The *anatomical preprocessing fast-track*: a new experimental command-line option
```

File: CHANGES.rst [hunk 9]
```diff
@@ -1,25 +1,134 @@
...
+  ``--anat-derivatives <PATH>`` checks that all necessary anatomical derivatives
+  required by *fMRIPrep* are present under ``<PATH>``, and skips the anatomical
+  processing in full if *fMRIPrep*'s expectations are met.
+  Because now functional processing of many sessions and runs can be efficiently
+  split into more digestible computational units (i.e., cluster job) while guaranteeing the
+  exact same anatomical results are being used, this can significantly speed up
+  longitudinal study preprocessing, and it is a fundamental optimization to process
+  databases of densely scanned individuals such as `My Connectome
+  <https://openneuro.org/datasets/ds000031>`__.
```
Comment (mgxd) [inline CHANGES.rst:30]:
```suggestion
  <https://openneuro.org/datasets/ds000031>`__. This option is not recommended 
  for single session processing.
```

**Comment by mgxd** [comment]
ok, in it goes 😅

**Pull request merged**